### PR TITLE
cmake: cleanup

### PR DESCRIPTION
**Pull request by joergsteffens**
### Thank you for contributing to the Bareos Project!

During PR #1592 it turns out that a number of cmake config settings could be cleaner,
especially when it comes to macOS configuration.
Instead of integrating this in the bugfixing PR, this is now handled in this separate PR.

This PR replaces #1610, as only a PR in https://github.com/bareos/bareos can be used to test the new macOS build workflow handling.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Required backport PRs have been created
- [x] https://git.bareos.com/bareos/CD/pulls/64 merged?
- [x] `.github/workflows/build-macos.yml` available in bareos-23 (yes), bareos-22 (#1691) and bareos-21 (#1692)?

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

### CODE REVIEW

File: cmake/BareosSetVariableDefaults.cmake [hunk 1]
```diff
@@ -28,231 +28,176 @@ if(NOT DEFINED prefix)
   set(prefix ${CMAKE_DEFAULT_PREFIX})
 endif()
 
+option(USE_RELATIVE_PATHS
+       "Compile with relatives path, required for relocatable binaries." OFF
```
Comment (arogge) [inline cmake/BareosSetVariableDefaults.cmake:32]:
```suggestion
       "Compile with relative paths, required for relocatable binaries." OFF
```

File: core/src/lib/jcr.cc [hunk 2]
```diff
@@ -521,23 +510,24 @@ std::shared_ptr<JobControlRecord> GetJcrById(uint32_t JobId)
...
+  return GetJcr([name](const JobControlRecord* jcr) {
+    return std::string_view{jcr->Job}.find(name) == 0;
   });
 }
 
-std::shared_ptr<JobControlRecord> GetJcrBySession(const VolumeSessionInfo& vsi)
+std::shared_ptr<JobControlRecord> GetJcrBySession(VolumeSessionInfo vsi)
 {
-  return GetJcr([&vsi](const JobControlRecord* jcr) {
+  static_assert(sizeof(vsi) == 8);
```
Comment (arogge) [inline core/src/lib/jcr.cc:529]: why do we need to check that?

**Review by arogge — CHANGES_REQUESTED** [review]
Looks pretty good overall. I have a few questions and I think you should check if it makes sense to mark some of the cache variables as advanced, so they're hidden from the default-view in ccmake.

File: core/src/lib/jcr.cc (hunk 2 above)
Comment (sebsura) [reply core/src/lib/jcr.cc:529]: Just making sure that the struct does not get too big for pass-by-value.

**Comment by joergsteffens** [comment]
@arogge I applied your suggestion and also mark some of the variables as advanced.

**Review by arogge — APPROVED** [review]
Sorry for taking that long. This looks really, really good!

**Pull request merged**